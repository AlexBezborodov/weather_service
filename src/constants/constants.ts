export const API_KEY = "1db2eafd57c395fd6ee79ed8ab422f37";
export const BASE_URL = `http://api.openweathermap.org/data/2.5/weather?`;
export const MULTIPLE_CITIES_URL = `http://api.openweathermap.org/data/2.5/group?id=`;
