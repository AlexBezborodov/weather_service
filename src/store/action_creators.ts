import { CityData, ErrorData } from "../interfaces";
import {
  GET_CITY_WEATHER,
  UPDATE_CITY_WEATHER,
  DELETE_CITY_WEATHER,
  ADD_SAVED_CITIES,
} from "./action_type_constants";

export const getCityWeather = (payload: CityData | ErrorData) => ({
  type: GET_CITY_WEATHER,
  payload,
});

export const updateCityWeather = (payload: CityData | ErrorData) => ({
  type: UPDATE_CITY_WEATHER,
  payload,
});
export const deleteCity = (payload: string | number) => ({
  type: DELETE_CITY_WEATHER,
  payload,
});
export const addSavedCitiesData = (payload: any) => ({
  type: ADD_SAVED_CITIES,
  payload,
});
