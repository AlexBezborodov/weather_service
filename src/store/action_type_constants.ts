export const GET_CITY_WEATHER = "GET_CITY_WEATHER";
export const UPDATE_CITY_WEATHER = "UPDATE_CITY_WEATHER";
export const DELETE_CITY_WEATHER = "DELETE_CITY_WEATHER";
export const FETCH_SAVED_CITIES_DATA = "FETCH_SAVED_CITIES_DATA";
export const ADD_SAVED_CITIES = "ADD_SAVED_CITIES";
