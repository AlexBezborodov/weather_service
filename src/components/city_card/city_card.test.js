import React from "react";

import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";

import { TEST_CITY } from "../../constants";
import { store } from "../../store/store";
import CityCard from "./city_card";

describe("CityCard component", () => {
  it("CityCard renders", () => {
    render(
      <Provider store={store}>
        <Router>
          <CityCard city={TEST_CITY} />
        </Router>
      </Provider>
    );
    expect(screen.getByText("Zocca")).toBeInTheDocument();
  });
  it("CityCard snapshot", () => {
    const item = render(
      <Provider store={store}>
        <Router>
          <CityCard city={TEST_CITY} />
        </Router>
      </Provider>
    );
    expect(item).toMatchSnapshot();
  });
});
