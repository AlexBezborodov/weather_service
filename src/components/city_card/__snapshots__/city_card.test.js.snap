// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityCard component CityCard snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-y1osw4-MuiPaper-root-MuiCard-root"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
            >
              Zocca
            </div>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Country: 
              IT
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Temperature: 
              + 25
               째C /
               
              + 78
               째F
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
            >
              Weather: 
              Rain
            </p>
            <div>
              <img
                alt="Weather icon"
                id="wicon"
                src="http://openweathermap.org/img/w/10d.png"
              />
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiCardActions-root MuiCardActions-spacing css-1fjc5kp-MuiCardActions-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-1knaqv7-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Refresh weather
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeSmall MuiButton-textSizeSmall css-6ekzlo-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Delete
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-y1osw4-MuiPaper-root-MuiCard-root"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
          >
            Zocca
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Country: 
            IT
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Temperature: 
            + 25
             째C /
             
            + 78
             째F
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          >
            Weather: 
            Rain
          </p>
          <div>
            <img
              alt="Weather icon"
              id="wicon"
              src="http://openweathermap.org/img/w/10d.png"
            />
          </div>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiCardActions-root MuiCardActions-spacing css-1fjc5kp-MuiCardActions-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-1knaqv7-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Refresh weather
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeSmall MuiButton-textSizeSmall css-6ekzlo-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Delete
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
